contracts/infrastructure/DappRegistry.sol

```diff
@@ -2,77 +2,197 @@ pragma solidity ^0.6.12;
+    // authorised dapps and their filters for each registry id
+    mapping (uint8 => mapping (address => bytes32)) public authorisations; // [registryId] => [dapp] => [{filter:160}{validAfter:64}]
+    // pending authorised dapps and their filters for each registry id
+    mapping (uint8 => mapping (address => bytes32)) public pendingFilterUpdates; // [registryId] => [dapp] => [{filter:160}{validAfter:64}]
+    // owners for each registry id
+    mapping (uint8 => address) public registryOwners; // [registryId] => [owner]
     
-    // mapping [wallet][name][status]
-    mapping (address => mapping (bytes32 => bool)) public isEnabled;
-    // mapping [name][address][authorisation]
-    mapping (bytes32 => mapping (address => Authorisation)) public authorisations;
+    event RegistryCreated(uint8 registryId, address registryOwner);
+    event RegistryRemoved(uint8 registryId);
+    event OwnerChanged(uint8 registryId, address newRegistryOwner);
+    event TimelockChangeRequested(uint64 newSecurityPeriod);
+    event TimelockChanged(uint64 newSecurityPeriod);
+    event FilterUpdated(uint8 indexed registryId, address dapp, address filter, uint256 validAfter);
+    event FilterUpdateRequested(uint8 indexed registryId, address dapp, address filter, uint256 validAfter);
+    event DappAdded(uint8 indexed registryId, address dapp, address filter, uint256 validAfter);
+    event DappRemoved(uint8 indexed registryId, address dapp);
 
-    bytes32[] public registries;
+    modifier onlyOwner(uint8 _registryId) {
+        validateOwner(_registryId);
+        _;
+    }
+    
+    constructor(uint64 _securityPeriod) public {
+        // set the timelock period
+        securityPeriod = _securityPeriod;
+        // set the owner of the Argent Registry (registryId == 0)
+        registryOwners[0] = msg.sender;
+    }
 
-    mapping (bytes32 => address) public registryManager;
+    /********* Wallet-centered functions *************/
 
-    event RegistryCreated(bytes32 registry, address manager);
-    event RegistryRemoved(bytes32 registry);
+    function isEnabledRegistry(address _wallet, uint8 _registryId) external view returns (bool) {
+        uint registries = uint(enabledRegistryIds[_wallet]);
+        return ((registries >> _registryId) & 1) > 0;
+    }
 
-    function authorise(address _wallet, address _contract, bytes calldata _data) external view override returns (bool) {
-        (bool isActive, address filter) = getFilter(_wallet, _contract);
+    function isAuthorised(address _wallet, address _spender, address _to, bytes calldata _data) external view override returns (bool) {
+        (bool isActive, address filter) = getFilter(_wallet, _spender);
         if (isActive) {
-            return _data.length == 0 || filter == address(0) || IFilter(filter).validate(_data);
+            return _data.length == 0 || filter == address(0) || IFilter(filter).validate(_wallet, _spender, _to, _data);
```

Comment:
I think we should remove the condition _data.length == 0 and let the filter decide if it accepts ETH transfers.